sdk/storage/Azure.Storage.DataMovement/src/StreamToUriJobPart.cs

```diff
@@ -195,64 +195,61 @@ public override async Task ProcessPartToChunkAsync()
                     return;
                 }
                 await OnTransferStateChangedAsync(TransferState.InProgress).ConfigureAwait(false);
-                long? fileLength = default;
-                StorageResourceItemProperties properties = await _sourceResource.GetPropertiesAsync(_cancellationToken).ConfigureAwait(false);
-                fileLength = properties.ResourceLength;
 
-                if (fileLength.HasValue)
+                StorageResourceItemProperties sourceProperties =
+                    await _sourceResource.GetPropertiesAsync(_cancellationToken).ConfigureAwait(false);
+                if (!sourceProperties.ResourceLength.HasValue)
```

Comment:
Diff is ugly here since I pulled this if out in front and un-indented everything below. Also, some cleanup changes to make it a little easier to read and to match `ServiceToServiceJobPart` a little more